### PR TITLE
Add e2e validation test for kine

### PR DESCRIPTION
**Pull request by vitorsavian**
#### Proposed Changes ####

- E2E validation for kine

#### Types of Changes ####

- Test

#### Verification ####

- See CI

#### User-Facing Change ####

```release-note

```

#### Further Comments ####

- Why a new test and not with `validateCluster`? My main thought was that the validateCluster test was made for rke2 without experimental features, so kine needs its own place.

### CODE REVIEW

**Review by VestigeJ — APPROVED** [review]
LGTM nice

File: tests/e2e/kine/kine_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	"github.com/rancher/rke2/tests/e2e"
+)
+
+// Valid nodeOS: bento/ubuntu-24.04, opensuse/Leap-15.6.x86_64
+var nodeOS = flag.String("nodeOS", "bento/ubuntu-24.04", "VM operating system")
+var serverCount = flag.Int("serverCount", 2, "number of server nodes")
+var agentCount = flag.Int("agentCount", 1, "number of agent nodes")
+var ci = flag.Bool("ci", false, "running on CI")
+
+// Environment Variables Info:
```
Comment (dereknola) [inline tests/e2e/kine/kine_test.go:21]: Add a line here about `E2E_EXTERNAL_DB` as that is the primary variable that will change with this test.

File: tests/e2e/kine/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,157 @@
+ENV['VAGRANT_NO_PARALLEL'] = ENV['E2E_STANDUP_PARALLEL'] ? nil : 'no'
+NODE_ROLES = (ENV['E2E_NODE_ROLES'] ||
+  ["server-0", "server-1", "agent-0"])
+NODE_BOXES = (ENV['E2E_NODE_BOXES'] ||
+  ['bento/ubuntu-24.04', 'bento/ubuntu-24.04', 'bento/ubuntu-24.04', 'bento/ubuntu-24.04'])
```
Comment (dereknola) [inline tests/e2e/kine/Vagrantfile:5]:
```suggestion
  ['bento/ubuntu-24.04', 'bento/ubuntu-24.04', 'bento/ubuntu-24.04'])
```
There are more boxes than roles defined.

**Review by dereknola — CHANGES_REQUESTED** [review]

File: tests/e2e/kine/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,157 @@
+ENV['VAGRANT_NO_PARALLEL'] = ENV['E2E_STANDUP_PARALLEL'] ? nil : 'no'
+NODE_ROLES = (ENV['E2E_NODE_ROLES'] ||
+  ["server-0"])
```
Comment (ShylajaDevadiga) [inline tests/e2e/kine/Vagrantfile:3]: Let's change the default to 3 server if we want to run with three servers

Comment (dereknola) [reply tests/e2e/kine/Vagrantfile]: You can't run it in GHA then

Comment (dereknola) [reply tests/e2e/kine/Vagrantfile]: The nightly should set the variables to different values, with the GHA running the default of one.

**Review by dereknola — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**